tests/compile.js

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env node
+
+// Recompile baseline ink files with the current version of inklecate avilable
+// $PATH.
+
+let childProcess = require('child_process');
+let fs = require('fs-extra');
+let path = require('path');
+
+let fileDirectory = path.join(__dirname, 'files');
+let inkFileDirectory = path.join(fileDirectory, 'original');
+let compiledFileDirectory = path.join(fileDirectory, 'compiled');
+
+// These files require the `-c` flag so that all visits will
+// be counted.
+let filesRequiringCFlag = [
+	'visit_counts_when_choosing.ink',
+	'turns_since_with_variable_target.ink',
+	'read_count_variable_target.ink',
+	'tests.ink'
+]
+
+function getAllInkFiles(dir) {
+	return fs.readdirSync(dir).reduce((files, file) => {
+		if (fs.statSync(path.join(dir, file)).isDirectory()) {
+			// 'includes' directories are not processed since they are expected
+			// to be included by inklecate directly.
+			if (file !== 'includes') {
+				return files.concat(getAllInkFiles(path.join(dir, file)));
+			} else {
+				return files;
+			}
+		} else {
+			if (path.extname(file) === '.ink') {
+				return files.concat(path.join(dir, file));
+			} else {
+				return files;
+			}
+		}
+	}, []);
+}
+
+function runInklecate(input, output, extraArgs) {
+	let command = `inklecate ${extraArgs} -o "${output}" "${input}"`
```

Comment:
I wonder if we should use something like https://github.com/furkleindustries/inklecate-node instead of expecting `inklecate` to be in PATH? Or potentially include a version of inklecate in this repo if we want tight control over what version we use?

I personally need an ink compiler on a semi regular basis when working or debuging things in inkjs and usually end up downloading the latest version again. Having a compiler ready would come in handy.

Reply:
I took a very hands-off approach, putting the burden of configuring inklecate on the user. I maintain [homebrew taps for macOS](https://github.com/ephread/homebrew-ink) and I use the versioning capabilities to switch between different versions of inklecate (I would love if someone could do the same for linux and windows).

But since this is a JavaScript project, you're right, it makes a lot of sense to leverage the excellent work from furkle. The only downside I see is that we have no ways of switching between different inklecate versions or know which one is currently bundled. Maybe that could be fixed with a PR?

I think bundling `inklecate` on our own would be overkill though.

I'll have a deeper look at these options at the very end!

Reply:
> The only downside I see is that we have no ways of switching between different inklecate versions or know which one is currently bundled. Maybe that could be fixed with a PR?

Yeah this is my concern as well. i don't think we'd ever want to go back to older inklecate versions, but we'd probably want it to be relatively up to date.

@furkle Do you keep inkleate up to date on `inklecate-node`? Or would you accept PRs that do it?

> I'll have a deeper look at these options at the very end!

Yeah this isn't blocking at all, no problem.

Reply:
> i don't think we'd ever want to go back to older inklecate versions

Actually, I'm fairly confident furkle keeps it up to date. But, for inkgd, I juggled between 0.8.2, 0.8.3 and 0.9.0. I wanted to make sure that the current runtime worked with older versions of inklecate as well (as long as they shared the same inkVersion internally), that's what I meant :smile:

Might happen at some point with inkjs as well! Oh an maybe it'd be a good idea to have a matrix in the CI and test with multiple different versions of inklecate?

Reply:
>  I wanted to make sure that the current runtime worked with older versions of inklecate as well

You're right, that makes sense. A matrix in the CI would make sense to test several inklecates, but now I'm not sure how we'd have several versions to begin with. Except if we keep inklecate versions in the repo, but as you say it might be overkill.

Reply:
We can open an issue about this after 0.9.0!